profiler/data_collector.rst

```diff
@@ -73,6 +78,11 @@ The getters are added to give the template access to the collected information.
 
 .. caution::
 
+    If the data that is not directly related to the request or response, you need to make the data accessible to your DataCollector.
+    This can be achieved by injecting the service that holds the information you intend to profile into your DataCollector.
```

Comment:
I don't think a code example would improve anything here. At the point you're creating your own DataCollector, you're probably already aware of injection and DI. The comment is a nice addition as it is right now imo. In case people get stuck here, there's always IRC or Slack, where they will probably be forwarded to the DIC docs.